.github/ISSUE_TEMPLATE/spec_request.md

```diff
@@ -0,0 +1,108 @@
+  title: # insert a human-readable name of the format here. MANDATORY.
+  application: # insert a human-readable name of the application for which this format is native here. If it is native for several applications, choose the ones you feel the most affiliated with the format. If the format is a widespread one, like PNG or WAV, just remove this field.
+  file-extension:
+    # insert here a YAML list of file extensions should be placed here.
+  xref: # check if the following sources contain info about the format you request. See https://doc.kaitai.io/user_guide.html#meta-xref for more info
+    din:
+      id: # DIN standard identifier
+    forensicswiki: # Page name on https://forensicswiki.org/
+    fileformat: # Link to https://www.fileformat.info
+    iso: # ISO standard identifier
+    justsolve: # ID on http://fileformats.archiveteam.org/wiki/
+    loc: # ID in https://www.loc.gov/preservation/digital/formats/
+    mime: # MIME type
+    pronom: # IDs on https://www.nationalarchives.gov.uk/pronom/fmt/
+     - x-fmt/
+    rfc: # Link to RFC
+    wikidata: Q # ID on WikiData
+
+doc: |
+  # Describe your format briefly AND add links to the sources of sample files in the Web. MANDATORY
+
+doc-ref:  # A YAML array of URIs to information sources and/or their names goes here. MANDATORY. See https://doc.kaitai.io/ksy_reference.html#doc-ref for more info on sources which are not just an URI
+  - ref 1
+  - ref 2
+  - ref 3
+  # There are at least 3 kinds of sources complementing each other.
+  # * Official specs.
+  #    * Pros:
+  #      * they are official
+  #      * they usually conduct the the meaning by the authors of the format.
+  #    * Cons:
+  #      * often obsolete and incomplete - don't document extensions by third-parties, even de-facto standards, met in the wild
+  #      * often they are licensed under a proprietary license, so it may be illegal to quote them extensively into `doc`.
+  #    * How to find:
+  #      * usually in PDF, HTML and RTF, Google dork filetype:pdf may be useful
+  # * Third-party specs. They are often either reverse-engineered or combined from info of other people specs.
+  #    * Pros:
+  #      * often more complete and up-to-date
+  #      * often available under license permitting reuse
+  #    * Cons:
+  #      * unofficial
+  #    * How to find:
+  #      * Usually reside in GitHub, Gist.GitHub and different MediaWiki websites.
+  #      * Usually in txt, MarkDown, asciidoc, ReStucturedText and MediaWiki formats.
+  # * Free software source code
+  #    * Pros:
+  #      * gives insight into poorly documented and/or unofficial details.
+  #    * Cons:
+  #      * you have to reverse-engineer meaning from the code
+  #      * the license is often viral, like GPL
```

Comment:
Again, "often" really depends on what formats you're dealing with. There is a lot of permissively-licensed free software. Aside from that, IANAL, but you can probably use GPL code as a reference without affecting the spec (as long as you don't, for example, translate a GPLed header one-to-one into a Kaitai spec).

Reply:
>but you can probably use GPL code as a reference without affecting the spec (as long as you don't, for example, translate a GPLed header one-to-one into a Kaitai spec).

not sure in that, I am not a lawer.

Reply:
>There is a lot of permissively-licensed free software

And there are lot of GPLd one. For example I currently am implementing a spec for R serialization format (I wanna read it from python, but the existing packages require R installed, I don't want that shit in my system), but R is GPLd and I haven't managed to find an impl using a permissive license. And the format is also completely undocumented.